l10n_th_account_tax/models/account_move.py

```diff
@@ -188,7 +188,8 @@ def write(self, vals):
                         }
                     )
                     line.tax_invoice_ids |= taxinv
-            else:
+            # Unlink all tax invoice, when manual_tax_invoice change from True to False
+            elif self.manual_tax_invoice and vals["manual_tax_invoice"] is False:
```

Comment:
Why use 'elif', and not just 'else' ?

Reply:
If I use else during the update, there will be some Journal Items lines that, when recomputed, will send manual_tax_invoice = false.

![image](https://github.com/user-attachments/assets/b5092dd2-1b59-426a-96f2-1cb09f46da16)

Which will cause "self.mapped("tax_invoice_ids").unlink()" to execute and delete the tax invoice even though the value has not been updated from true to false

Reply:
@kittiu